broker-daemon/utils/base64-lex.spec.js

```diff
@@ -0,0 +1,110 @@
+  })
+
+  it('preserves lexicographic ordering for numbers vs uppercase letters', () => {
+    base64LowerIndex = 'A' // base64 index for 'A' is 0
+    base64HigherIndex = '0' // base64 index for '0' is 52
+
+    // ASCII character codes would incorrectly sort based on base64 for this case
+    expect(base64LowerIndex.charCodeAt(0)).to.be.greaterThan(base64HigherIndex.charCodeAt(0))
+
+    lexLowerIndex = encodeBase64Lex(base64LowerIndex)
+    lexHigherIndex = encodeBase64Lex(base64HigherIndex)
+
+    expect(lexLowerIndex.charCodeAt(0)).to.be.lessThan(lexHigherIndex.charCodeAt(0))
+  })
+
+  it('preserves lexicographic ordering for numbers vs lowercase letters', () => {
+    base64LowerIndex = 'a' // base64 index for 'a' is 26
+    base64HigherIndex = '0' // base64 index for '0' is 52
+
+    // ASCII character codes would incorrectly sort based on base64 in this case
+    expect(base64LowerIndex.charCodeAt(0)).to.be.greaterThan(base64HigherIndex.charCodeAt(0))
+
+    lexLowerIndex = encodeBase64Lex(base64LowerIndex)
+    lexHigherIndex = encodeBase64Lex(base64HigherIndex)
+
+    expect(lexLowerIndex.charCodeAt(0)).to.be.lessThan(lexHigherIndex.charCodeAt(0))
+  })
+
+  it('preserves lexicographic ordering for `/`', () => {
+    base64LowerIndex = '0' // base64 index for '0' is 52
+    base64HigherIndex = '/' // base64 index for '/' is 63
+
+    // ASCII character codes would incorrectly sort based on base64 for this case
+    expect(base64LowerIndex.charCodeAt(0)).to.be.greaterThan(base64HigherIndex.charCodeAt(0))
+
+    lexLowerIndex = encodeBase64Lex(base64LowerIndex)
+    lexHigherIndex = encodeBase64Lex(base64HigherIndex)
+
+    expect(lexLowerIndex.charCodeAt(0)).to.be.lessThan(lexHigherIndex.charCodeAt(0))
+  })
+
+  it('preserves lexicographic ordering for `+`', () => {
+    base64LowerIndex = '0' // base64 index for '0' is 52
+    base64HigherIndex = '+' // base64 index for '+' is 62
+
+    // ASCII character codes would incorrectly sort based on base64 for this case
+    expect(base64LowerIndex.charCodeAt(0)).to.be.greaterThan(base64HigherIndex.charCodeAt(0))
+
+    lexLowerIndex = encodeBase64Lex(base64LowerIndex)
+    lexHigherIndex = encodeBase64Lex(base64HigherIndex)
+
+    expect(lexLowerIndex.charCodeAt(0)).to.be.lessThan(lexHigherIndex.charCodeAt(0))
+  })
+
+  it('preserves lexicographic ordering for data used as order IDs', () => {
+    // When the following timestamps are base64 encoded, they lose their ordering by timestamp
+    // Using encodeBase64Lex preserves the ordering by timestamp
+    const timestamps = [
+      1554914185000,
+      1554505000000,
+      1554831515000
+    ]
+
+    const expected = [
+      { timestamp: 1554505000000, lexId: 'M9US9-' },
+      { timestamp: 1554831515000, lexId: 'M9nNak' },
+      { timestamp: 1554914185000, lexId: 'M9sQXF' }
+    ]
+
+    // Convert timestamps to base64 encoding then convert to lex ordering
+    const lexIds = []
+    timestamps.forEach(ts => {
```

Comment:
why not just map timestamps to an array of `lexIds`? Then you can just do `.sort()` without a custom sort function.